### DISCUSSION TITLE
Inconsistent Node.js version upgrades

### DISCUSSION THREAD

**Discussion by Turbo87** [renovatebot/renovate#12592] [Request Help]
In a Node.js project I'm having the Node.js version specified in three different files:

- `package.json` in the `engines.node` field pinned as `14.18.1`
- `Dockerfile` via `FROM node:14.18.1-alpine`
- `.github/workflows/ci.yml` via `NODE_VERSION: 14.18.1` env var and using the `regex` manager for renovate

My problem is:

- the `package.json` file is updating only to Node.js LTS releases (regex manager is configured to behave similarly)
- the `Dockerfile` update uses a different datasource and `versioning` (due to the `-alpine` suffix) and tries to update to 17.x

I would ideally like to only get a single PR that updates all three files to the same version, but I've not found a way to do this so far. Even if I configure the `package.json` update to use `semver` versioning to avoid the LTS restriction it suddenly starts including a `v` prefix in the version... 😅 

Any help with this would be appreciated. The repository where I'm having this issue is https://github.com/rust-lang/crates.io

**Comment by rarkins** [comment]
We created a `node` versioning specifically to understand which releases were "stable" (LTS) or not. The challenge is that when you use an alpine tag in Docker you need to use our `docker` versioning, which knows nothing about node LTS. I'm guessing e.g. if you just used `node:14.18.1` in Dockerfile and set to use the `node` versioning then you'd get what you expected.

I think what we need is something new, along similar lines to `extractVersion`, where you can break a version into two bits: the actual version.. and compatibility. Then the `-alpine` is compatibility while the `14.18.1` can still be processed using `node` versioning.

**Reply by Turbo87** [reply]
> I think what we need is something new, along similar lines to extractVersion, where you can break a version into two bits: the actual version.. and compatibility. Then the -alpine is compatibility while the 14.18.1 can still be processed using node versioning.

Yep, exactly. If we could mark the `-alpine` as not being part of the actual version string it would simplify the problem :)

**Reply by rarkins** [reply]
I think we're on the same page. It would mean e.g. `docker` versioning is possibly not even needed any more, e.g. we'd split docker tags into the version bit (could use loose versioning or coerced semver) and the compatibility bit
